openhcl/shared_pool_alloc/src/lib.rs

```diff
@@ -266,6 +267,11 @@ impl user_driver::vfio::VfioDmaBuffer for SharedPoolAllocator {
             .map_file(0, len, gpa_fd.get(), file_offset, true)
             .context("unable to map allocation")?;
 
+        // SAFETY: The previous call to memmap should have made exactly len bytes at offset 0 valid to read and write
```

Comment:
Can we add a comment about _why_ we're doing the zeroing? Perhaps we should update the VfioDmaBuffer trait to note that zeroing out these pages is a requirement.